### PR TITLE
Modernize a bit

### PR DESCRIPTION
**Pull request by Goooler**
Just all things.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1 +1,32 @@
+pluginManagement {
+    repositories {
+        google()
+        gradlePluginPortal()
+    }
+}
+
+// https://docs.gradle.com/enterprise/gradle-plugin/
+plugins {
+    id "com.gradle.enterprise" version "3.12.2"
```
Comment (connyduck) [inline settings.gradle:10]: I don't think we need this plugin

File: app/src/main/java/com/keylesspalace/tusky/MainActivity.kt [hunk 2]
```diff
@@ -506,8 +506,8 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
                         startActivityWithSlideInAnimation(AnnouncementsActivity.newIntent(context))
                     }
                     badgeStyle = BadgeStyle().apply {
-                        textColor = ColorHolder.fromColor(MaterialColors.getColor(binding.mainDrawer, R.attr.colorOnPrimary))
-                        color = ColorHolder.fromColor(MaterialColors.getColor(binding.mainDrawer, R.attr.colorPrimary))
+                        textColor = ColorHolder.fromColor(MaterialColors.getColor(binding.mainDrawer, com.google.android.material.R.attr.colorOnPrimary))
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/MainActivity.kt:509]: What difference do these changes make?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -50,8 +49,15 @@ photoview = "2.3.0"
 sparkbutton = "4.1.0"
 unified-push = "2.0.1"
 
+[plugins]
+android-application = { id = "com.android.application", version.ref = "agp" }
+android-library = { id = "com.android.library", version.ref = "agp" }
```
Comment (connyduck) [inline gradle/libs.versions.toml:54]: Not used anywhere?

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+agp = "7.4.0"
+androidx-activity = "1.6.1"
+androidx-appcompat = "1.6.0"
 androidx-browser = "1.4.0"
 androidx-cardview = "1.0.0"
 androidx-constraintlayout = "2.1.4"
 androidx-core = "1.9.0"
 androidx-exifinterface = "1.3.4"
-androidx-fragment = "1.5.3"
+androidx-fragment = "1.5.5"
```
Comment (connyduck) [inline gradle/libs.versions.toml:10]: AndroidX dependencies are already being upgraded in https://github.com/tuskyapp/Tusky/pull/3169

**Review by connyduck — COMMENTED** [review]
Nice, but a a lot of mixed changes for one PR

File: app/src/main/java/com/keylesspalace/tusky/MainActivity.kt (hunk 2 above)
Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/MainActivity.kt:509]: https://developer.android.com/studio/past-releases/past-agp-releases/agp-4-1-0-release-notes#4.1-nontransitive-r-class

File: gradle/libs.versions.toml (hunk 3 above)
Comment (Goooler) [reply gradle/libs.versions.toml:54]: We can remove `android-library`.

File: gradle/libs.versions.toml (hunk 4 above)
Comment (Goooler) [reply gradle/libs.versions.toml]: f84c5b980e78da12c4828dbad8aa4e59ed37f02c

File: settings.gradle (hunk 1 above)
Comment (Goooler) [reply settings.gradle]: f3be6d01a2231b1134f341f8f708c306496628da

File: gradle/libs.versions.toml (hunk 3 above)
Comment (Goooler) [reply gradle/libs.versions.toml]: 10aa548081f9554edd7b27dabd0332dc0e1f6340

File: app/src/main/java/com/keylesspalace/tusky/MainActivity.kt (hunk 2 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/MainActivity.kt:509]: Oh learned something new, thx

**Review by connyduck — APPROVED** [review]

**Comment by Goooler** [comment]
@connyduck Ready to merge?

**Comment by connyduck** [comment]
Yes, I'll merge it after Tusky 21 is released

**Pull request merged**